vllm/distributed/device_communicators/cuda_communicator.py

```diff
@@ -162,6 +198,50 @@ def destroy(self):
             self.all2all_manager.destroy()
             self.all2all_manager = None
 
+    def all_gatherv(self,
+                    input_: Union[torch.Tensor, list[torch.Tensor]],
+                    dim: int = 0,
+                    sizes: Optional[list[int]] = None):
+        if dim != 0:
+            raise NotImplementedError("only dim 0 all-gatherv is supported")
+        world_size = self.world_size
+        pynccl_comm = self.pynccl_comm
+        assert pynccl_comm is not None and not pynccl_comm.disabled
+
+        def _all_gather_single(input_: torch.Tensor,
+                               sizes: Optional[list[int]] = None):
+            input_size = input_.size()
+            if sizes is not None:
+                assert len(sizes) == world_size
+                assert input_.shape[dim] == sizes[self.rank_in_group]
+                output_size = (sum(sizes), ) + input_size[1:]
+                # 'sizes' is not needed if all inputs in the same group have the
+                # same shape
+                if all(s == sizes[0] for s in sizes):
+                    sizes = None
```

Comment:
This special case seems like it could be calculated in all_gatherv before calling _all_gather_single